src/Development/IDE/Core/Rules.hs

```diff
@@ -253,51 +252,71 @@ getLocatedImportsRule =
-                          -> (a, Maybe ArtifactsLocation)
-                          -> ((PathIdMap, IntMap ArtifactsLocation), (a, Maybe FilePathId))
-                        f (pathMap, ss) (imp, mbPath) = case mbPath of
-                            Nothing -> ((pathMap, ss), (imp, Nothing))
-                            Just path ->
-                                let (pathId, pathMap') = getPathId path pathMap
-                                    ss' = if isBootLocation path
-                                            then IntMap.insert (getFilePathId pathId) path ss
-                                            else ss
-                                in ((pathMap', ss'), (imp, Just pathId))
-                    -- Convert paths in imports to ids and update the path map
-                    let ((pathIdMap, ss'), modImports') = mapAccumL f (rawPathIdMap rawDepInfo, ss) modImports
-                    -- Files that we haven’t seen before are added to the queue.
-                    let newFiles =
-                            IntSet.fromList (coerce $ mapMaybe snd modImports')
-                            IntSet.\\ IntMap.keysSet (rawImports rawDepInfo)
-                    let rawDepInfo' = insertImport fId (Right $ ModuleImports modImports' pkgImports) rawDepInfo
-                    go (newFiles `IntSet.union` fs)
-                       (rawDepInfo' { rawPathIdMap = pathIdMap }, ss')
-
-
+    go :: [NormalizedFilePath] -- ^ Queue of paths to process
+       -> (RawDependencyInformation, IntMap ArtifactsLocation)
+            -- The accumulator we add information about the imports to
+       -> Action ([FilePathId] -- The return FilePathIds of the FilePath from the first argument
+                 , RawDependencyInformation, IntMap ArtifactsLocation)
+    go [] (r, i) = pure ([], r, i) -- Nothing to process, return the accumulator.
+    go (f:fs) a@(rawDepInfo, ss)
+      -- First check to see if we have already processed the FilePath
+      -- If we have, just return its Id but don't update any of the state.
+      | Just fid <- lookupPathToId (rawPathIdMap rawDepInfo) f = do
+          (fids, r, bms) <- go fs a
+          return (fid: fids, r, bms)
+      -- Otherwise, we need to process its imports.
+      | otherwise = do
+          al <- modSummaryToArtifactsLocation f <$> use_ GetModSummary f
+          -- Try to parse the imports of the file
+          importsOrErr <- use GetLocatedImports f
+          -- Get a fresh FilePathId for the new file
+          let (fId, path_map) = getPathId al (rawPathIdMap rawDepInfo)
+          -- Insert the File into the bootmap if it's a boot module
+          let ss' = if isBootLocation al
+                      then IntMap.insert (getFilePathId fId) al ss
+                      else ss
+          let rawDepInfo' = rawDepInfo { rawPathIdMap = path_map }
+          case importsOrErr of
+            Nothing -> do
+            -- File doesn’t parse so add the module as a failure into the
```

Comment:
The use of smart quotes in comments in source bugs me. You know this will cause a unicode issue one day, and all for a pretty comment...

Reply:
I’m happy to break setups that still cannot handle unicode. It’s 2020 :slightly_smiling_face: 
This comment is also not new (and probably introduced by me) and it hasn’t caused any issues in the last year or however long it was here.